packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -508,6 +520,24 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
     return FlutterCommandResult.success();
   }
 
+  Iterable<String> _getCoveragePackages(
+    List<String> packagesRegExps,
+    FlutterProject flutterProject,
+    PackageConfig packageConfig,
+  ) {
+    final String projectName = flutterProject.manifest.appName;
+    final Iterable<String> packagesToInclude;
+    if (packagesRegExps.isNotEmpty) {
+      final RegExp combinedRegExp = RegExp('(${packagesRegExps.join(")|(")})');
+      packagesToInclude = packageConfig.packages
```

Comment:
shouldn't we include `projectName` too?

Reply:
I believe `packageConfig.packages` include current package as well.

Reply:
but that's not guaranteed to match the provided regexes, right?

Reply:
Yes. I thought about `coverage-package` option as a generic option that defaults to the current package, so we can exclude current package from the report if we want to. But that is not necessary of course. 
So if we wanna always include the current package, I can update the code. What do you think?

Reply:
Hmm, I see what you mean. At least to my mind, this is counter-intuitive behavior, because I assume you would always want to include the current package. Can you think of a use case when you would NOT want coverage of the current package? Otherwise this behavior would just mean users would always have to specify another two arguments to include the current package.

Reply:
Either way we go I think we should make it explicit in the help text.

Reply:
> Can you think of a use case when you would NOT want coverage of the current package?

This is what comes to my mind:
For plugins/packages, we usually have an `example` app inside package (e.g. [camera_android](https://github.com/flutter/packages/tree/main/packages/camera/camera_android) plugin), where most of the tests are located. When launching tests from the `example` app and collecting coverage, what we really want is to collect coverage for the package itself, and we don't really care (I think) about the code inside the `example`.

Anyway, I don't think it would be harmful to have coverage data for the example as well.

Reply:
I've updated PR to always include current package, but we can continue discussion

Reply:
sorry to flip flop, but per discussion in the discord, I think the way you originally had it, *without* the current package is probably better

Reply:
Current package is included only by default in the new changes